Jenkinsfile

```diff
@@ -564,13 +564,27 @@ pipeline
                         {
                             unstash 'uaa-war'
                         }
-                        
-                        sh """
-                            cp build/cloudfoundry-identity-uaa-*.war iam-container-config/uaa/cloudfoundry-identity-uaa.war
-                            cd iam-container-config/uaa/
-                            docker build --no-cache -t uaa:${artifactVersion} -f Dockerfile .
-                            docker images
-                        """
+                        script {
+                            String OTEL_JAR_NAME = "splunk-otel-javaagent.jar"
+                            String OTEL_EXTENSION_REPO = "artifactory.build.ge.com"
+                            String OTEL_EXTENSION_VERSION = "1.1.0.RELEASE"
+                            String OTEL_EXTENSION_PATH = "/artifactory/APM-AWS/com/ge/apm/ged-opentelemetry-java-extension/${OTEL_EXTENSION_VERSION}/"
+                            String OTEL_EXTENSION_JAR_NAME="ged-opentelemetry-java-extension-${OTEL_EXTENSION_VERSION}.jar"
+                            withCredentials([usernamePassword(credentialsId: 'BUILD_GE_ARTIFACTORY_CREDENTIALS',
+                                    usernameVariable: 'BUILDGE_USER', passwordVariable: 'BUILDGE_PSWRD')]) {
+                                sh """
+                                    cp build/cloudfoundry-identity-uaa-*.war iam-container-config/uaa/cloudfoundry-identity-uaa.war
+
+                                    cd iam-container-config/uaa/
+
+                                    curl -L https://github.com/signalfx/splunk-otel-java/releases/latest/download/splunk-otel-javaagent.jar -o $OTEL_JAR_NAME
+                                    curl --user ${BUILDGE_USER}:${BUILDGE_PSWRD} https://${OTEL_EXTENSION_REPO}${OTEL_EXTENSION_PATH}${OTEL_EXTENSION_JAR_NAME} -o $OTEL_EXTENSION_JAR_NAME
+
+                                    docker build --build-arg="OTEL_JAR_NAME=${OTEL_JAR_NAME}" --build-arg="OTEL_EXTENSION_JAR_NAME=${OTEL_EXTENSION_JAR_NAME}" --no-cache -t uaa:${artifactVersion} -f Dockerfile .
```

Comment:
Its nice passing the file names in because the extension file name includes the version, so if we need to upgrade that we only have to change this jenkinsfile (and hopefully just the version variable).

Reply:
Alternatively, we could have moved the jar to one named without the version, I think this is simpler by avoiding the copy and I like preserving the version (in the filename) in the image.